### PR TITLE
Arithmetic tests

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: host/xtest/regression_4100.c [hunk 1]
```diff
@@ -0,0 +1,2536 @@
...
+			goto out;
+	}
+out:
+	TEEC_CloseSession(&session);
+}
+ADBG_CASE_DEFINE(regression, 4106, test_4106,
+		"Test TEE Internal API Arithmetical API - Neg");
+
+static bool do_mul(ADBG_Case_t *c, TEEC_Session *s, const char *str_s,
+		      const char *str_t, const char *str_r)
```
Comment (etienne-lms) [inline host/xtest/regression_4100.c]: few minor indentation issues: some manual space indent that do not match.
Check also definition header for `test_4110_mod()` and `test_4110_fmm()`.

**Comment by jenswi-linaro** [comment]
Addressed all comments but the one about "CHECK_PT()"

**Comment by jenswi-linaro** [comment]
Update

**Comment by jforissier** [comment]
@etienne-lms ping?

File: host/xtest/regression_4100.c [hunk 2]
```diff
@@ -0,0 +1,2539 @@
...
+	/* ha = ha * hb */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, cmd_mul(c, s, ha, hb, ha)))
+		goto out;
+	/* check ha == hr */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, compare_handle(c, s, ha, hr, 0)))
+		goto out;
+
+	res = true;
+out:
+	if (ha != TA_CRYPT_ARITH_INVALID_HANDLE)
```
Comment (etienne-lms) [inline host/xtest/regression_4100.c]: minor: condition could be removed.

File: host/xtest/regression_4100.c [hunk 3]
```diff
@@ -0,0 +1,2539 @@
...
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, cmd_new_var(c, s, 1024, &hb)))
+		goto out;
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, cmd_new_var(c, s, 1024, &hc)))
+		goto out;
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, cmd_new_var(c, s, 1024, &hr)))
+		goto out;
+
+	/* First do the ADD dominated phase */
+
+
```
Comment (etienne-lms) [inline host/xtest/regression_4100.c]: minor: 2 empty lines

File: host/xtest/regression_4100.c [hunk 4]
```diff
@@ -0,0 +1,2539 @@
...
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+				cmd_cmp_s32(c, s, hr, 0, &cres)))
+			goto out;
+		if (!ADBG_EXPECT_COMPARE_SIGNED(c, cres, ==, rsign))
+			goto out;
+	}
+
+	res = true;
+out:
+	if (hn != TA_CRYPT_ARITH_INVALID_HANDLE)
```
Comment (etienne-lms) [inline host/xtest/regression_4100.c]: minor: condition could be removed.

Few other occurrences below.

**Comment by etienne-lms** [comment]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`
Minor things that could be addressed. Very clean test integration.

File: host/xtest/regression_4100.c (hunk 2 above)
Comment (jenswi-linaro) [reply host/xtest/regression_4100.c]: I'll fix all threse

File: host/xtest/regression_4100.c (hunk 3 above)
Comment (jenswi-linaro) [reply host/xtest/regression_4100.c]: I'll fix all these

**Comment by jenswi-linaro** [comment]
Comments addressed, tags applied, squashed and rebased.
I've restarted the build of https://github.com/OP-TEE/optee_os/pull/2682 to see that everything builds and tests OK

**Comment by jenswi-linaro** [comment]
Tested OK on Hikey, added a tag for that too.

**Pull request merged**